### PR TITLE
demo of project organization

### PR DESCRIPTION
**Pull request by nolanlawson**
OK, so this isn't a literal pull request, but I just wanted to show you what I managed to do with a little closed-source demo app I wrote a few months ago. (Can't open-source the app, but I can open-source these scripts and general architecture.)

Most of this code is generated by Ionic by running `ionic start myApp sidemenu` (see [their guide](http://ionicframework.com/getting-started/)). I don't think we necessarily want to hitch ourselves to Ionic, but they are built on top of Cordova, and there might be value in making this thing basically just a Cordova add-on.

The most interesting part of this PR are:
- the `bin/` directory, which has scripts to package up the Cordova app into a Node-Webkit app, an Appcache-using webapp, and a Crosswalk app
- all the `PUT_ICONS_HERE.txt` files. Look at all the friggin' boilerplate you have to add just to have icons in the right spot for iOS, Android, FirefoxOS, and Mac apps. (For the other platforms, I cannot guarantee that the icons are in the right spot; I didn't test them.)

As far as Crosswalk goes, I am pretty convinced that in 2015 any decent HTML5 bundler for Android ought to build using Crosswalk, due to the performance issues [I brought up in my talk](http://nolanlawson.github.io/brooklyn-js-html5-mobile-apps). Furthermore building separate APKs for arm/x86 is not actually that difficult; I've got some instructions [here](http://forum.ionicframework.com/t/crosswalk-integration-beta-in-ionic-cli-v1-3-2/15190/48), and in our case we could do it with shell scripts instead of needing a full Gradle setup.

So in short, I am really excited about this project, but man oh man, it's going to be a lot of work. :smiley: And it's probably important to choose a good architecture right from the start, to make things easier. Cordova + shell scripts might not be optimal, but it worked pretty well for this little one-off project.

### CODE REVIEW

File: merges/firefoxos/manifest.webapp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "512": "/icon/icon-512.png",
+    "128": "/icon/icon-128.png",
+    "60": "/icon/icon-60.png"
+  },
+  "developer": {
+    "name": "Me",
+    "url": "http://me.com"
+  },
+  "default_locale": "en"
+}
```
Comment (nolanlawson) [inline merges/firefoxos/manifest.webapp:15]: I have no idea why you need a separate format for FirefoxOS compared to iOS and Android, but I suspect it's because the Cordova tools for FirefoxOS just aren't there yet. :/

File: icons/PUT_ICONS_HERE.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+e.g. these icons:
+
+android-hdpi.png    icon-bb-150.png  ios-40.png     ios-57.png	   ios-72.png	  wp-99.png
+android-mdpi.png    icon-bb-86.png   ios-40@2x.png  ios-57@2x.png  ios-72@2x.png
+android-xhdpi.png   ios-29.png	     ios-50.png     ios-60.png	   ios-76.png
+android-xxhdpi.png  ios-29@2x.png    ios-50@2x.png  ios-60@2x.png  ios-76@2x.png
```
Comment (nolanlawson) [inline icons/PUT_ICONS_HERE.txt:6]: These should probably be moved to the `www/` folder so they can be re-used for the webapp (i.e. using the `<link rel="apple-touch-icon"` stuff), but I didn't get that far.

**Comment by reggi** [comment]
@nolanlawson Should I have not pulled it in? Eek! I can revert.

Duh. first line...

> Ok, so this isn't a literal pull request

**Comment by nolanlawson** [comment]
Heh yeah, I mostly just wanted to send you a diff. No biggie, though; we can delete stuff we don't need. :)

In retrospect, I think it may be nicer if we didn't enforce a particular project structure on users of `oneapp`. Ideally they should just be able to provide HTML/CSS/JS plus icons/name/description, and under the hood we could use Cordova or whatever.

**Comment by reggi** [comment]
Gotcha! I reverted I think it makes more sense to not confuse people that this isn't a project yet and it's more of a discussion / proof of concept.
